app/lib/advanced_text_formatter.rb

```diff
@@ -7,9 +7,10 @@ def initialize(options, &block)
       @format_link = block
     end
 
-    def block_code(code, _language)
+    def block_code(code, language)
+      # Looks wrong, but sets title to language correctly. One downside is, it adds an empty attribute when no lang specified.
       <<~HTML
-        <pre><code>#{ERB::Util.h(code).gsub("\n", '<br/>')}</code></pre>
+        <pre><code data-codelang="#{language}">#{ERB::Util.h(code).gsub("\n", '<br/>')}</code></pre>
```

Comment:
I don't suppose `language` is guaranteed to be escaped?

Reply:
I'm not sure what kind of escaping would be suitable here.
At least in my testing I couldn't cause any unexpected behaviour, nevertheless I agree accepting user input as is is never a good idea.
Ideally, I would check here if `language` is a valid code language, but this would create overhead.
What I could do though is checking that `language` only contains allowed characters via regex: `/^[a-zA-Z0-9.+-]*$/`